### PR TITLE
CCP-356: make sure preview text appears in html and design emails

### PR DESCRIPTION
**Pull request by zawadzkip**
- Updates the insert preview text function to insert for HTML based emails also
- updates the insert preview text "hack" based on the article mentioned

## Testing
- Tested against local AD serve with html & design type emails

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Review by anusontarangkul — APPROVED** [review]
Looks good to me

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts [hunk 1]
```diff
@@ -237,8 +237,7 @@ export class SendEmailPerformer extends MessageSendPerformer<Settings, Payload>
       )
     }
 
-    // only include preview text in design editor templates
-    if (this.payload.bodyType === 'design' && this.payload.previewText) {
+    if (this.payload.previewText) {
       const parsedPreviewText = await this.parseTemplating(this.payload.previewText, { profile }, 'Preview text')
```
Comment (zawadzkip) [inline packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:241]: Wrap in a try-catch, if something fails lets log it and move on with the original HTML

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @zawadzkip and @anusontarangkul , 
Do you want this deployed today? Are you happy with the PR?

**Comment by zawadzkip** [comment]
@joe-ayoub-segment yeah! that would be great

**Comment by joe-ayoub-segment** [comment]
Hi @zawadzkip , I didn't hear back from you soon enough unfortunately, sorry! 
Also there is a flakey test which is causing the CI for this PR to fail. Could you please pull the latest files from main into this PR? The same thing happed to a number of other PRs yesterday.

**Comment by zawadzkip** [comment]
I see an error happening in my test now, updating

**Comment by zawadzkip** [comment]
@joe-ayoub-segment good to go

**Comment by joe-ayoub-segment** [comment]
Hi @zawadzkip this PR has been deployed.

**Pull request merged**